transport-native-io_uring/src/main/java/io/netty/channel/uring/IOUringSubmissionQueue.java

```diff
@@ -188,17 +188,33 @@ public boolean addClose(int fd) {
     public void submit() {
         int submit = tail - head;
         if (submit > 0) {
-            PlatformDependent.putIntOrdered(kTailAddress, tail); // release memory barrier
-            int ret = Native.ioUringEnter(ringFd, submit, 0, 0);
-            head = PlatformDependent.getIntVolatile(kHeadAddress); // acquire memory barrier
-            if (ret != submit) {
-                if (ret < 0) {
-                    throw new RuntimeException("ioUringEnter syscall");
-                }
-                logger.warn("Not all submissions succeeded");
+            submit(submit, 0, 0);
+        }
+    }
+
+    public void submitAndWait() {
+        int submit = tail - head;
+        if (submit > 0) {
+            submit(submit, 1, Native.IORING_ENTER_GETEVENTS);
+        } else {
+            int ret = Native.ioUringEnter(ringFd, 0, 1, Native.IORING_ENTER_GETEVENTS);
```

Comment:
Guarding for negative `submit` counts? When would that happen?

Reply:
I think that would entail a nasty bug but I guess it wouldn't harm to check.

Reply:
On second thoughts I don't think there's really a need to "guard" for this here, since we still take the "correct" action i.e. don't submit and wait for completions. But I will add a check to where we trigger submit when the SQ ring is full, since if for some reason that returns 0 we should not move to the next SQE slot.

Reply:
Yeah, that's my thinking. The dance _seems_ unnecessary, but I'm not entirely certain. Things like overflow is hard to reason about.

Reply:
Good candidate for assert?